app/src/main/java/com/ataulm/artcollector/gallery/ui/GalleryActivity.kt

```diff
@@ -40,12 +45,30 @@ class GalleryActivity : AppCompatActivity() {
             adapter.submitList(gallery)
         })
 
-        viewModel.events.observe(this, EventObserver {
-            val intent = when (it) {
-                is NavigateToArtistGallery -> artistGalleryIntent(it.artist.id)
-                is NavigateToPainting -> paintingIntent(it.painting.artist.id, it.painting.id)
+        viewModel.events.observe(this, EventObserver { command ->
+            when (command) {
+                is NavigateToArtistGallery -> navigateToArtistGallery(command)
+                is NavigateToPainting -> navigateToPainting(command)
             }
-            startActivity(intent)
         })
     }
+
+    private fun navigateToArtistGallery(it: NavigateToArtistGallery) {
+        val intent = artistGalleryIntent(it.artist.id)
+        startActivity(intent)
+    }
+
+    private fun navigateToPainting(command: NavigateToPainting) {
+        val (painting, view) = command.painting to command.view
+        val paintingIntent = paintingIntent(painting.artist.id, painting.id, painting.imageUrl)
+        val heroImage = Pair(view, getString(R.string.shared_element_painting))
+        val options = ActivityOptionsCompat.makeSceneTransitionAnimation(this, heroImage)
+        startActivity(paintingIntent, options.toBundle())
```

Comment:
these three lines:

```kotlin
val heroImage = Pair(view, getString(R.string.shared_element_painting))
val options = ActivityOptionsCompat.makeSceneTransitionAnimation(this, heroImage)
startActivity(paintingIntent, options.toBundle())
```

is basically how to do the shared element transition. The other required piece is setting `android:transitionName` in the source & target XML layouts for the image view.

Reply:
commit 7449e7fa3ea452f5a40fd236969bf94d047ebe9b (HEAD -> shared-element-transition, origin/shared-element-transition)
Author: Ataul Munim <hello@ataulm.com>
Date:   Sat Nov 24 12:42:24 2018 +0000

    Remove passing of View to ViewModel
    
    - rely instead on adapterPosition so that we don't have to pass ImageView to the ViewModel
    - this solves the issue of the memory leak by keeping a reference to activity context in the ViewModel's livedata

Reply:
commit 64c848de02b7647e2f9f4003584e4b54c912e32b
Author: Ataul Munim <hello@ataulm.com>
Date:   Sat Nov 24 12:56:31 2018 +0000

    Replace Painting/Artist onClicks with adapterPosition
    
    - since we _need_ the view itself to do the transition, then it's not so much that we care about what the
    user clicked on in terms of domain (i.e. this/that painting) AS LONG AS we can reconcile it
    - in these terms, the adapterPosition is the unique identifier so long as the adapter contents remains the
    same from the point of clicking to the point of resolving the click